execution/aggregate/scalar_table.go

```diff
@@ -163,34 +174,78 @@ func makeAccumulatorFunc(expr parser.ItemType, arg parser.Expr) (newAccumulatorF
 			var hasValue bool
 
 			return &accumulator{
-				AddFunc: func(v float64) {
+				AddFunc: func(_ uint64, v float64) {
 					if !hasValue {
 						value = v
 					} else {
 						value = math.Min(value, v)
 					}
 					hasValue = true
 				},
-				ValueFunc: func() float64 { return value },
-				HasValue:  func() bool { return hasValue },
+				ValueFunc: func(_ uint64) float64 { return value },
+				HasValue:  func(_ uint64) bool { return hasValue },
 				Reset: func() {
 					hasValue = false
 					value = 0
 				},
 			}
 		}, nil
+	case "topk":
+		return func() *accumulator {
+			k := int(arg.(*parser.NumberLiteral).Val)
```

Comment:
I think `k` here needs to come from a `model.VectorOperator` and be iterated over somehow. This would fail for scalar expressions that contain binary operators or select from storage. For eg `topk(3-2, http_requests_total)` would have an interface conversion error here.

Reply:
Oh..I see..

This means that we already have this problem here?

https://github.com/thanos-community/promql-engine/blob/1a53dd6b99d8da0502361dcd14437ad18bb098ec/execution/aggregate/scalar_table.go#L286-L287

Reply:
Yeah, nice catch, it's the same problem there too!

Reply:
Created an issue:

https://github.com/thanos-community/promql-engine/issues/108

Idk if we should go ahead with this PR as it is or holt it until this issue get fixed.

WDYT?

Reply:
We don't need to block this PR! 🙂  
If this can be fixed here with maybe a new operator that can handle parameterized aggregations, quantile can be moved to that new operator then:  https://github.com/thanos-community/promql-engine/pull/107#issuecomment-1299612683